src/libraries/System.Linq/tests/SequenceTests.cs

```diff
@@ -0,0 +1,301 @@
+                Assert.Equal([T.MaxValue], Enumerable.Sequence(T.MaxValue, T.MaxValue, T.One));
+                Assert.Equal([T.MaxValue], Enumerable.Sequence(T.MaxValue, T.MaxValue, T.MaxValue));
+            }
+
+            void ValidateSigned<T>() where T : INumber<T>, IMinMaxValue<T>
+            {
+                ValidateUnsigned<T>();
+
+                Assert.Equal([T.One, T.Zero, T.CreateTruncating(-1), T.CreateTruncating(-2), T.CreateTruncating(-3), T.CreateTruncating(-4)], Enumerable.Sequence(T.One, T.CreateTruncating(-4), T.CreateTruncating(-1)));
+                Assert.Equal([T.One, T.CreateTruncating(-1), T.CreateTruncating(-3)], Enumerable.Sequence(T.One, T.CreateTruncating(-4), T.CreateTruncating(-2)));
+                Assert.Equal([T.One, T.CreateTruncating(-2)], Enumerable.Sequence(T.One, T.CreateTruncating(-4), T.CreateTruncating(-3)));
+                Assert.Equal([T.One, T.CreateTruncating(-3)], Enumerable.Sequence(T.One, T.CreateTruncating(-4), T.CreateTruncating(-4)));
+                Assert.Equal([T.One, T.CreateTruncating(-4)], Enumerable.Sequence(T.One, T.CreateTruncating(-4), T.CreateTruncating(-5)));
+                Assert.Equal([T.One], Enumerable.Sequence(T.One, T.CreateTruncating(-4), T.CreateTruncating(-6)));
+
+                Assert.Equal([T.MinValue], Enumerable.Sequence(T.MinValue, T.MinValue, T.CreateTruncating(-11)));
+            }
+        }
+
+        [Fact]
+        public void FloatingPoint_ProduceExpectedSequence()
+        {
+            float[] results;
+
+            results = Enumerable.Sequence(0.123f, 0.456f, 0.123f).ToArray();
+            Assert.Equal(3, results.Length);
+            Assert.Equal(0.123f, results[0], 3);
+            Assert.Equal(0.246f, results[1], 3);
+            Assert.Equal(0.369f, results[2], 3);
+
+            results = Enumerable.Sequence(0.123f, -0.456f, -0.123f).ToArray();
+            Assert.Equal(5, results.Length);
+            Assert.Equal(0.123f, results[0], 3);
+            Assert.Equal(0.0f, results[1], 3);
+            Assert.Equal(-0.123f, results[2], 3);
+            Assert.Equal(-0.246f, results[3], 3);
+            Assert.Equal(-0.369f, results[4], 3);
+
+            results = Enumerable.Sequence(16_777_216f, float.MaxValue, 1.0f).ToArray();
+            Assert.Equal(1, results.Length);
+            Assert.Equal(16_777_216f, results[0], 3);
+
+            results = Enumerable.Sequence(-16_777_217f, float.MinValue, -1.0f).ToArray();
+            Assert.Equal(1, results.Length);
+            Assert.Equal(-16_777_216f, results[0], 3);
+        }
+
+        [Fact]
+        public void SmallIntegers_ProducesFullRange()
+        {
+            byte[] bytes = Enumerable.Sequence(byte.MinValue, byte.MaxValue, (byte)1).ToArray();
+            for (int i = 0; i < bytes.Length; i++)
```

Comment:
Do you think it's worth checking the length as well as the positions?  E.g. `Assert.Equal(256, bytes.Length);` (applies to all 4 types here, and the same in the async)